TodoList/TodoList.Api.Tests/Controllers/NodesControllerTest.cs

```diff
@@ -24,18 +25,52 @@ internal class NodeControllerTest
 
         public NodesController Controller;
 
+        private INodesRepository MockRepository()
+        {
+            var repository = Substitute.For<INodesRepository>();
+
+            repository.GetAllAsync().Returns(new[]
+            {
+                new NodeModel {Id = FirstId, Text = "poopy"},
+                new NodeModel {Id = SecondId, Text = "GEARS"},
+                new NodeModel {Id = ThirdId, Text = "Planet Music"},
+                new NodeModel {Id = FourthId, Text = "Time to get shwifty"}
+            });
+
+            repository.GetByIdAsync(FirstId)
+                .Returns(new NodeModel {Id = FirstId, Text = "poopy"});
+
+            repository.AddAsync(new NodeModel()).ReturnsForAnyArgs(new NodeModel
+            {
+                Id = SecondId,
+                Text = "GEARS"
+            });
+
+            repository.UpdateAsync(new NodeModel()).ReturnsForAnyArgs(new NodeModel
+            {
+                Id = ThirdId,
+                Text = "Planet Music"
+            });
+
+            return repository;
+        }
+
         [SetUp]
         public void SetUp()
         {
-            Controller = GetControllerForTests();
+            Controller = GetControllerForTests(MockRepository());
         }
 
-        private static NodesController GetControllerForTests()
+        private static NodesController GetControllerForTests(INodesRepository repository)
```

Comment:
It would be more consistent if you have method `MockLocationHelper` and in SetUp, you call both "Mock" methods and create the controller.